azure-cs-aks-private-container-registry/Program.cs

```diff
@@ -0,0 +1,216 @@
+				Build = "./SampleApplication",
+				Registry = new ImageRegistry {
+					Server = registry.LoginServer,
+					Username = registry.AdminUsername,
+					Password = registry.AdminPassword
+				},
+				ImageName = imageName
+			}, new ComponentResourceOptions {
+				Provider = k8sProvider
+			});
+
+			// Create a k8s secret for use when pulling images from the container registry when deploying the sample application.
+			var dockerCfg = Output.All<string>(registry.LoginServer, registry.AdminUsername, registry.AdminPassword).Apply(values => {
+				var r = new Dictionary<string, object>();
+				var server = values[0];
+				var username = values[1];
+				var password = values[2];
+
+				r[server] = new {
+					email = "notneeded@notneeded.com",
+					username,
+					password
+				};
+
+				return r;
+			});
+
+			var dockerCfgString = dockerCfg.Apply(x =>
+				Convert.ToBase64String(Encoding.UTF8.GetBytes(System.Text.Json.JsonSerializer.Serialize(x))));
+
+			var dockerCfgSecretName = "dockercfg-secret";
+
+			var dockerCfgSecret = new Pulumi.Kubernetes.Core.V1.Secret(dockerCfgSecretName, new SecretArgs {
+				Data = {
+					{ ".dockercfg", dockerCfgString }
+				},
+				Type = "kubernetes.io/dockercfg",
+				Metadata = new ObjectMetaArgs {
+					Name = dockerCfgSecretName,
+				}
+			}, new CustomResourceOptions {
+				Provider = k8sProvider
+			});
+
+			// Deploy the sample application to the cluster.
+			var labels = new InputMap<string> {
+				{ "app", $"app-{applicationName}" },
+			};
+
+			var deployment = new Pulumi.Kubernetes.Apps.V1.Deployment(applicationName, new DeploymentArgs {
+				Spec = new DeploymentSpecArgs {
+					Selector = new LabelSelectorArgs {
+						MatchLabels = labels,
+					},
+					Replicas = 1,
+					Template = new PodTemplateSpecArgs {
+						Metadata = new ObjectMetaArgs {
+							Labels = labels,
+							Name = applicationName
+						},
+						Spec = new PodSpecArgs {
+							Containers = new List<ContainerArgs> {
+								new ContainerArgs {
+									Name = applicationName,
+									Image = image.ImageName,
+								}
+							},
+							ImagePullSecrets = new LocalObjectReferenceArgs {
+								Name = dockerCfgSecretName
+							}
+						}
+					}
+				}
+			}, new CustomResourceOptions {
+				Provider = k8sProvider
+			});
+
+			// Create a new service.
+			var service = new Pulumi.Kubernetes.Core.V1.Service(applicationName, new ServiceArgs {
+				Metadata = new ObjectMetaArgs {
+					Name = applicationName,
+					Labels = labels
+				},
+				Spec = new ServiceSpecArgs {
+					Type = "LoadBalancer",
+					Selector = deployment.Spec.Apply(x => x.Template.Metadata.Labels),
+					Ports = new ServicePortArgs {
+						Port = 80
+					}
+				}
+			}, new CustomResourceOptions {
+				Provider = k8sProvider
+			});
+
+			return new Dictionary<string, object?> {
+				["kubeconfig"] = cluster.KubeConfigRaw,
+				["dockercfg-secret-name"] = dockerCfgSecret.Metadata.Apply(x => x.Name),
```

Comment:
How would I use this output?

Reply:
You would use it when creating a `Deployment` from another service project/stack when you want to push & pull a docker container image there from the registry created in this stack. This sample project uses it itself on line 184:

```csharp
ImagePullSecrets = new LocalObjectReferenceArgs {
	Name = dockerCfgSecretName
}
```

Depending on one's setup, probably the "horizontal" stack wouldn't be deploying any pods to the cluster - that would be done from "vertical" stacks - in which case the exported docker registry container secret comes handy.

Let me know if I should clarify this with a comment or something, or if you prefer I do something else.

Reply:
Maybe add a short comment to the README about this, thank you!